driver/docker/docker.go

```diff
@@ -301,9 +302,37 @@ func (d *Driver) fetchOutputs(ctx context.Context, container string, op *driver.
 		return opResult, err
 	}
 
+	// if an applicable output is expected but does not exist and it has a
+	// non-empty default value, create an entry in the map with the
+	// default value as its contents
+	for name, output := range op.Bundle.Outputs {
+		if !existsInOutputsMap(opResult.Outputs, name) && output.AppliesTo(op.Action) {
+			if outputDefinition, exists := op.Bundle.Definitions[output.Definition]; exists {
+				outputDefault := outputDefinition.Default
+				if outputDefault != nil {
+					filepath := unix_path.Join("/cnab", "app", "outputs", name)
+					contents := fmt.Sprintf("%v", outputDefault)
+					opResult.Outputs[filepath] = contents
+				} else {
+					return opResult, fmt.Errorf("required output %s is missing and has no default", name)
+				}
+			}
+		}
+	}
+
 	return opResult, nil
 }
 
+func existsInOutputsMap(outputsMap map[string]string, output string) bool {
+	var exists bool
+	for outputPath := range outputsMap {
+		if strings.Contains(outputPath, output) {
+			exists = true
```

Comment:
I guess you can early return here? `return true`, and at the end `return false` ?